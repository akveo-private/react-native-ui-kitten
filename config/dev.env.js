const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
