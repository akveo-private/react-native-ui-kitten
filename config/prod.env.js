const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
