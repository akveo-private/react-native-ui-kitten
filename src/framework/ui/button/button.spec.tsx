import React from 'react';
import {
  ImageSourcePropType,
  TouchableOpacity,
} from 'react-native';
import {
  render,
  fireEvent,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  Button as ButtonComponent,
  Props as ButtonProps,
} from './button.component';
import * as config from './button.spec.config';

const Button = styled<ButtonComponent, ButtonProps>(ButtonComponent);

const Mock = (props?: ButtonProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <Button {...props} />
  </StyleProvider>
);

const renderComponent = (props?: ButtonProps): RenderAPI => {
  return render(
    <Mock {...props} />,
  );
};

describe('@button: matches snapshot', () => {

  describe('* interaction', () => {

    it('* stateless', () => {
      const component: RenderAPI = renderComponent();
      const { output } = shallow(component.getByType(ButtonComponent));

      expect(output).toMatchSnapshot();
    });

  });

  describe('* appearance', () => {

    const icon: ImageSourcePropType = { uri: 'https://akveo.github.io/eva-icons/fill/png/128/star.png' };
    const text: React.ReactText = 'BUTTON';

    it('* empty', () => {
      const component: RenderAPI = renderComponent();
      const { output } = shallow(component.getByType(ButtonComponent));

      expect(output).toMatchSnapshot();
    });

    it('* icon', () => {
      const component: RenderAPI = renderComponent({ icon });
      const { output } = shallow(component.getByType(ButtonComponent));

      expect(output).toMatchSnapshot();
    });

    it('* text', () => {
      const component: RenderAPI = renderComponent({ children: text });
      const { output } = shallow(component.getByType(ButtonComponent));

      expect(output).toMatchSnapshot();
    });

    it('* icon and text', () => {
      const component: RenderAPI = renderComponent({
        icon,
        children: text,
      });
      const { output } = shallow(component.getByType(ButtonComponent));

      expect(output).toMatchSnapshot();
    });

  });

});

describe('@button: component checks', () => {

  it('* emits onPress', () => {
    const onPress = jest.fn();

    const component: RenderAPI = renderComponent({
      onPress: onPress,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(onPress).toBeCalled();
  });

});
