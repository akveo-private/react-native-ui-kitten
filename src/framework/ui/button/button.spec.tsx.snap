// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@button: matches snapshot * appearance * empty 1`] = `
<TouchableOpacity
  activeOpacity={1}
  alignment="left"
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  size="medium"
  status="primary"
  style={
    Array [
      Object {
        "borderRadius": 6,
        "flexDirection": "row",
        "minHeight": 24,
        "minWidth": 24,
        "padding": 4,
      },
      undefined,
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
      "text-primary": "#000000",
      "text-primary-inverse": "#ffffff",
    }
  }
/>
`;

exports[`@button: matches snapshot * appearance * icon 1`] = `
<TouchableOpacity
  activeOpacity={1}
  alignment="left"
  appearance="default"
  dispatch={[Function]}
  icon={
    Object {
      "uri": "https://akveo.github.io/eva-icons/fill/png/128/star.png",
    }
  }
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  size="medium"
  status="primary"
  style={
    Array [
      Object {
        "borderRadius": 6,
        "flexDirection": "row",
        "minHeight": 24,
        "minWidth": 24,
        "padding": 4,
      },
      undefined,
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
      "text-primary": "#000000",
      "text-primary-inverse": "#ffffff",
    }
  }
>
  <Component
    source={
      Object {
        "uri": "https://akveo.github.io/eva-icons/fill/png/128/star.png",
      }
    }
    style={
      Array [
        Object {
          "height": 14,
          "marginHorizontal": 4,
          "tintColor": "#ffffff",
          "width": 14,
        },
        Object {},
      ]
    }
  />
</TouchableOpacity>
`;

exports[`@button: matches snapshot * appearance * icon and text 1`] = `
<TouchableOpacity
  activeOpacity={1}
  alignment="left"
  appearance="default"
  dispatch={[Function]}
  icon={
    Object {
      "uri": "https://akveo.github.io/eva-icons/fill/png/128/star.png",
    }
  }
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  size="medium"
  status="primary"
  style={
    Array [
      Object {
        "borderRadius": 6,
        "flexDirection": "row",
        "minHeight": 24,
        "minWidth": 24,
        "padding": 4,
      },
      undefined,
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
      "text-primary": "#000000",
      "text-primary-inverse": "#ffffff",
    }
  }
>
  <Component
    source={
      Object {
        "uri": "https://akveo.github.io/eva-icons/fill/png/128/star.png",
      }
    }
    style={
      Array [
        Object {
          "height": 14,
          "marginHorizontal": 4,
          "tintColor": "#ffffff",
          "width": 14,
        },
        Object {},
      ]
    }
  />
  <ForwardRef(createComponent)
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "800",
          "marginHorizontal": 4,
        },
        Object {},
      ]
    }
  >
    BUTTON
  </ForwardRef(createComponent)>
</TouchableOpacity>
`;

exports[`@button: matches snapshot * appearance * text 1`] = `
<TouchableOpacity
  activeOpacity={1}
  alignment="left"
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  size="medium"
  status="primary"
  style={
    Array [
      Object {
        "borderRadius": 6,
        "flexDirection": "row",
        "minHeight": 24,
        "minWidth": 24,
        "padding": 4,
      },
      undefined,
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
      "text-primary": "#000000",
      "text-primary-inverse": "#ffffff",
    }
  }
>
  <ForwardRef(createComponent)
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "800",
          "marginHorizontal": 4,
        },
        Object {},
      ]
    }
  >
    BUTTON
  </ForwardRef(createComponent)>
</TouchableOpacity>
`;

exports[`@button: matches snapshot * interaction * stateless 1`] = `
<TouchableOpacity
  activeOpacity={1}
  alignment="left"
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  size="medium"
  status="primary"
  style={
    Array [
      Object {
        "borderRadius": 6,
        "flexDirection": "row",
        "minHeight": 24,
        "minWidth": 24,
        "padding": 4,
      },
      undefined,
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
      "text-primary": "#000000",
      "text-primary-inverse": "#ffffff",
    }
  }
/>
`;
